src/operator/switch.ts

```diff
@@ -6,20 +6,43 @@ import {InnerSubscriber} from '../InnerSubscriber';
 import {subscribeToResult} from '../util/subscribeToResult';
 
 /**
- * Converts an Observable that emits Observables into an Observable that emits the items emitted by the most recently
- * emitted of those Observables.
+ * Converts a higher-order Observable into a first-order Observable by only the
+ * most recently emitted of those nested Observables.
+ *
+ * <span class="informal">Flattens an Observable-of-Observables by dropping the
+ * previous nested Observable once a new one appears.</span>
  *
  * <img src="./img/switch.png" width="100%">
  *
- * Switch subscribes to an Observable that emits Observables. Each time it observes one of these emitted Observables,
- * the Observable returned by switchOnNext begins emitting the items emitted by that Observable. When a new Observable
- * is emitted, switchOnNext stops emitting items from the earlier-emitted Observable and begins emitting items from the
- * new one.
+ * `switch` subscribes to an Observable that emits Observables,
+ * also known as a higher-order Observable. Each time it observes one of these
+ * emitted nested Observables, the output Observable begins emitting the items
+ * emitted by that nested Observable. So far, it behaves like {@link mergeAll}.
+ * However, when a new nested Observable is emitted, `switch` stops emitting
+ * items from the earlier-emitted nested Observable and begins emitting items
+ * from the new one. It continues to behave like this for subsequent nested
+ * Observables.
```

Comment:
Will it makes sense to mention `switch` also handles subscription when switching occurs? I remember someone have asked if it's needed to explicitly manage inner subscription after switching.